includes/AddOns/TypesForSteelDesign/BoundaryCondition.js

```diff
@@ -0,0 +1,546 @@
+ * @param {Number} release_warping      Release of warping, can be undefined (is not set, false as default)
+ */
+SteelDesignBoundaryCondition.prototype.MemberHingesForStart = function (release_in_y,
+    release_about_x,
+    release_about_z,
+    release_warping) {
+    setSteelDesignBoundaryConditionMemberHinges(this.boundary_condition, 1, release_in_y, release_about_x, release_about_z, release_warping);
+};
+
+/**
+ * Sets Hinge releases for end node sequence
+ * @param {Number} release_in_y         Release in y, can be undefined (is not set, false as default)
+ * @param {Number} release_about_x      Release about x, can be undefined (is not set, false as default)
+ * @param {Number} release_about_z      Release about z, can be undefined (is not set, false as default)
+ * @param {Number} release_warping      Release of warping, can be undefined (is not set, false as default)
+ */
+SteelDesignBoundaryCondition.prototype.MemberHingesForEnd = function (release_in_y,
+    release_about_x,
+    release_about_z,
+    restraint_warping) {
+    setSteelDesignBoundaryConditionMemberHinges(this.boundary_condition, this.boundary_condition.member_hinges.row_count(), release_in_y, release_about_x, release_about_z, restraint_warping);
+};
+
+/**
+ * Sets member hinge intermediate node
+ * @param {Number} row                  Row (intermediate node begins on index 2)
+ * @param {Number} release_in_y         Release in y, can be undefined (is not set, false as default)
+ * @param {Number} release_about_x      Release about x, can be undefined (is not set, false as default)
+ * @param {Number} release_about_z      Release about z, can be undefined (is not set, false as default)
+ * @param {Number} release_warping      Release of warping, can be undefined (is not set, false as default)
+ */
+SteelDesignBoundaryCondition.prototype.SetMemberHingeIntermediateNode = function (row,
+    release_in_y,
+    release_about_x,
+    release_about_z,
+    release_warping) {
+    ASSERT(this.boundary_condition.intermediate_nodes, "Intermediate nodes must be on");
+    ASSERT(typeof row !== "undefined", "Row must be undefined");
+    setSteelDesignBoundaryConditionMemberHinges(this.boundary_condition, row, release_in_y, release_about_x, release_about_z, release_warping);
+};
+
+/**
+ * Sets Different properties state for member hinges
+ * @param {Boolean} different_properties_hinges     Different properties, can be undefined (true as default)
+ */
+SteelDesignBoundaryCondition.prototype.DifferentPropertiesForMemberHinges = function (different_properties_hinges) {
+    if (typeof different_properties_hinges === "undefined") {
+        different_properties_hinges = true;
+    }
+    this.boundary_condition.different_properties_hinges = different_properties_hinges;
+};
+
+/**
+ * @returns Nodal supports are set correctly
+ */
+SteelDesignBoundaryCondition.prototype.NodalSupportsEndEdit = function () {
```

Comment:
Confusing name of funciton
```suggestion
SteelDesignBoundaryCondition.prototype.IsSetNodalSupportsEnd = function () {
```